tremor-pipeline/src/op/generic/batch.rs

```diff
@@ -151,31 +151,33 @@ impl Operator for Batch {
         _state: &Value<'static>,
         signal: &mut Event,
     ) -> Result<EventAndInsights> {
-        self.max_delay_ns.map_or_else(
-            || Ok(EventAndInsights::default()),
-            |delay_ns| {
+        Ok(self
+            .max_delay_ns
+            .map_or_else(EventAndInsights::default, |delay_ns| {
                 if signal.ingest_ns - self.first_ns > delay_ns {
                     // We don't want to modify the original signal we clone it to
                     // create a new event.
 
                     let mut data = empty();
```

Comment:
could we move that down it feels like we don't need to swamp memory and allocate if there is nothing to swap